### PR TITLE
feat(component): enable the use of associations using strings

### PR DESCRIPTION
**Pull request by bas080**

### CODE REVIEW

**Comment by RWOverdijk** [comment]
Elaborate on this a bit please.

**Comment by doktordirk** [comment]
@bas080 meaning: what's the purpose of the propose changed

**Comment by bas080** [comment]
Association select's association and manyAssociation bindables use data binding to get and set values on the entity. The problem being that I'm not sure how to implement association select into Aurelia-form. It uses schema arrays containing element objects. How to perform databinding using those?

The changes made here are trying to solve this which till now has failed.

File: src/component/form-field.js [hunk 1]
```diff
@@ -18,7 +20,7 @@ export class FormField {
   constructor(config, viewManager) {
     this.config      = config;
     this.viewManager = viewManager;
-    this.model       = this;
+    this.vm          = this;
```
Comment (RWOverdijk) [inline src/component/form-field.js]: Use a name.

File: src/component/framework/association.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+export class Association {
```
Comment (RWOverdijk) [inline src/component/framework/association.js]: Add docblocks.

File: src/component/framework/association.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  association;
+  manyAssociation;
+
+  activate(vm) {
+    this.vm = vm; /* the form-fields view model */
```
Comment (RWOverdijk) [inline src/component/framework/association.js]: Use // for single line comments. Also, use a name in stead of vm.

File: src/aurelia-form.js [hunk 4]
```diff
@@ -23,6 +23,7 @@ export function configure(aurelia, configCallback) {
       'entity-form': './schema-form.html',
 
       /* custom elements with a view model do not end with .html */
+      association: '{{framepath}}/association',
```
Comment (RWOverdijk) [inline src/aurelia-form.js:26]: Align

File: src/component/form-field.html [hunk 5]
```diff
@@ -7,6 +7,6 @@
   <!-- has a view model this is a component with view model -->
   <compose
     if.bind="hasViewModel && visible"
-    model.bind="model"
+    model.bind="vm"
```
Comment (RWOverdijk) [inline src/component/form-field.html]: Use a name

File: src/aurelia-form.js (hunk 4 above)
Comment (bas080) [reply src/aurelia-form.js:26]: I will do this in another branch.

https://github.com/SpoonX/aurelia-form/issues/64

Comment (RWOverdijk) [reply src/aurelia-form.js:26]: Why?

Comment (bas080) [reply src/aurelia-form.js:26]: To avoid merge conflicts. I might have added or altered that object in other pull requests.

**Pull request merged**